pkg/db/src/implementations/version-1.2/customer-orders.ts

```diff
@@ -115,11 +143,89 @@ class CustomerOrder implements CustomerOrderInterface {
 		}, this.#initialized);
 	}
 
-	// get(): Promise<CustomerOrderInterface | undefined> {}
+	get(): Promise<CustomerOrderInterface | undefined> {
+		return runAfterCondition(async () => {
+			if (this.#exists) {
+				return this;
+			}
+
+			return Promise.resolve(undefined);
+		}, this.#initialized);
+	}
+
+	/**
+	 * Update customer order status (so far there's ordered and ready-for-pickup).
+	 * ready for pickup when all books have been received from supplier
+	 * (or some are received and others have to be re-ordered?)
+	 */
+	updateStatus(ctx: debug.DebugCtx, status: string): Promise<CustomerOrderInterface> {
+		const currentStatus = this.status;
+		debug.log(ctx, "customerOrder:updateStatus")({ status, currentStatus });
+
+		if (status === currentStatus || !status) {
+			debug.log(ctx, "customerOrder:updateStatus:noop")({ status, currentStatus });
+			return Promise.resolve(this);
+		}
+
+		debug.log(ctx, "customerOrder:updateStatus:updating")({ status });
+		return this.update(ctx, { status });
+	}
+
+	/**
+	 * Update individual book on customer order status.
+	 * only the client state (the supplier state is handled elsewere)
+	 * @TEMP book status: ordered, received or re-order
+	 */
+	updateBookStatus(ctx: debug.DebugCtx, isbn: string, status: string): Promise<CustomerOrderInterface> {
+		const index = this.books.findIndex((book) => book.isbn === isbn);
+		if (index === -1 || this.books[index].status === status) Promise.resolve(this);
+
+		const updatedBooks = [...this.books];
+		updatedBooks[index] = { ...updatedBooks[index], status };
+
+		return this.update(ctx, { books: updatedBooks });
+	}
```

Comment:
This is on the right track, however...let's support updating of multiple books. That way:
- we **can** update the entire batch - even if it's a subset of books ordered, it will still probably be multiple books a lot of the time
- if we need to update only one book - just to [book]